### PR TITLE
better handling provider tile update

### PR DESCRIPTION
**Pull request by heshan0131**
https://github.com/keplergl/kepler.gl/issues/990
Signed-off-by: Shan He <heshan0131@gmail.com>

### CODE REVIEW

File: src/components/modals/load-storage-map.js [hunk 1]
```diff
@@ -215,17 +218,23 @@ const VisualizationItem = ({vis, onClick}) => {
     </StyledVisualizationItem>
   );
 };
-const ProviderSelect = ({cloudProviders, onSetCloudProvider, currentProvider}) =>
+
+export const ProviderSelect = ({
+  cloudProviders = [],
+  onSelect,
+  onSetCloudProvider,
```
Comment (macrigiuseppe) [inline src/components/modals/load-storage-map.js]: is this still used?

File: src/components/modals/load-storage-map.js [hunk 2]
```diff
@@ -262,21 +277,38 @@ function LoadStorageMapFactory() {
...
       });
     }
+
+    _clickBack = () => {
+      this.setState({showProviderSelect: true});
+    };
+
+    _selectProvider = providerName => {
+      this.props.onSetCloudProvider(providerName);
+      const provider = (this.props.cloudProviders || []).find(p => p.name === providerName);
```
Comment (macrigiuseppe) [inline src/components/modals/load-storage-map.js]: what if provider is null?

I know we have a check in getSavedMaps but should we even dispatch the action if provider is undefined

Comment (heshan0131) [reply src/components/modals/load-storage-map.js]: `_selectProvider` is callbacks from `ProviderSelect` which is rendered by cloudProviders. I don't  see in which case provider will be underfined

File: src/components/modals/load-storage-map.js (hunk 1 above)
Comment (heshan0131) [reply src/components/modals/load-storage-map.js]: it is passed to `ProviderSelect`

**Pull request merged**